File: testdata/jennies/builders/builder_delegation/TypescriptBuilder/src/builder_delegation/dashboardlink_builder_gen.ts
import * as cog from '../cog';
import * as builder_delegation from '../builder_delegation';

export class DashboardLinkBuilder implements cog.Builder<builder_delegation.DashboardLink> {
    private readonly internal: builder_delegation.DashboardLink;

    constructor() {
        this.internal = builder_delegation.defaultDashboardLink();
    }

    build(): builder_delegation.DashboardLink {
        return this.internal;
    }

    title(title: string): this {
        this.internal.title = title;
        return this;
    }

    url(url: string): this {
        this.internal.url = url;
        return this;
    }
}

File: testdata/jennies/builders/builder_delegation/GoBuilder/builder_delegation/dashboardlink_builder_gen.go
package builder_delegation

import (
	cog "github.com/grafana/cog/generated/cog"
)

var _ cog.Builder[DashboardLink] = (*DashboardLinkBuilder)(nil)

type DashboardLinkBuilder struct {
    internal *DashboardLink
    errors cog.BuildErrors
}

func NewDashboardLinkBuilder() *DashboardLinkBuilder {
	resource := NewDashboardLink()
	builder := &DashboardLinkBuilder{
		internal: resource,
		errors: make(cog.BuildErrors, 0),
	}

	return builder
}



func (builder *DashboardLinkBuilder) Build() (DashboardLink, error) {
	if err := builder.internal.Validate(); err != nil {
		return DashboardLink{}, err
	}
	
	if len(builder.errors) > 0 {
	    return DashboardLink{}, cog.MakeBuildErrors("builder_delegation.dashboardLink", builder.errors)
	}

	return *builder.internal, nil
}

func (builder *DashboardLinkBuilder) Title(title string) *DashboardLinkBuilder {
    builder.internal.Title = title

    return builder
}

func (builder *DashboardLinkBuilder) Url(url string) *DashboardLinkBuilder {
    builder.internal.Url = url

    return builder
}

